versions/development/SPEC.md

```diff
@@ -302,7 +302,37 @@ For more details on the postfix quantifiers, see the section on [Optional Parame
 
 For more information on type and how they are used to construct commands and define outputs of tasks, see the [Data Types & Serialization](#data-types--serialization) section.
 
-#### Custom  Types
+#### Numeric Types
+
+`Int` and `Float` are the numeric types.
+Int can be used to hold any integer, there is no size limit.
+Float is used to hold a finite 64-bit IEEE-754 floating point number.
+Note that WDL itself is not intended to be used for heavy numeric computation.
+There are no number formatting utilities, many mathematical functions are absent,
+and engine behaviour at extreme values is not defined and should not be relied upon.
+
+The mathematical operators are defined only for identical types.
+In expressions using an operator with mismatched numeric types,
+the `Int` will be cast to `Float` and the result will be `Float`.
+Note that this will cause loss of precision. The `Float` can be
+converted to `Int` with the `ceil`, `round` or `floor` functions if needed.
+
+In expressions involving a numeric type and a String, the numeric type
+will be converted to a string. This is straigthforward for `Int`.
+For `Float`, behaviour will be similar to the 'g' conversion specifier
```

Comment:
I was wary to say 'g' as defined by C, as I was not sure how Java and Python would be compatible with that.

Reply:
I am not sure that referring to a certain behavior in a programming language at all is appropriate for the WDL specs. WDL was intended to usable by non-programmers, asking them to look things up about C, therefore, seems out of place/inappropriate.

It might be more appropriate and clearer to describe the expected behavior in full in the WDL specs themselves.